### PR TITLE
UMA-58 Official study material

### PR DESCRIPTION
**Pull request by trojanc**
This PR adds the content for the official study material

### CODE REVIEW

File: myadmin-angular/projects/myadmin-lib/src/lib/services/study-material.service.ts [hunk 1]
```diff
@@ -1,8 +1,9 @@
 import {Injectable} from "@angular/core";
 import {HttpClient} from "@angular/common/http";
 import {Observable} from "rxjs";
-import {StudentModuleEnrolmentInfo, StudyMaterialDetailInfo} from "../info-objects";
-import {StudentInfo} from "myadmin-lib";
+import {StudentInfo} from "../info-objects/student-info";
+import {StudentModuleEnrolmentInfo} from "../info-objects/student-module-enrolment-info";
```
Comment (jannie-louwrens) [inline myadmin-angular/projects/myadmin-lib/src/lib/services/study-material.service.ts]: I can't remember but I thought we agreed to dropping the Student prefix and rename it to be ModuleEnrolmentInfo.

File: myadmin-service-impls/integration-services/src/main/java/za/ac/unisa/myadmin/studymaterial/integration/services/StudyMaterialWebServiceClient.java [hunk 2]
```diff
@@ -72,21 +79,34 @@ public void setServiceTrustAllSsl(boolean serviceTrustAllSsl) {
...
 	private static int convertSemesterStringToId(String semester){
-		if (semester.equals("S1")) {
-			semester = "1";
-		} else if (semester.equals("S2")) {
-			semester = "2";
-		} else if (semester.equals("Y1")) {
-			semester = "0";
-		} else if (semester.equals("Y2")) {
-			semester = "6";
+		switch (semester) {
```
Comment (jannie-louwrens) [inline myadmin-service-impls/integration-services/src/main/java/za/ac/unisa/myadmin/studymaterial/integration/services/StudyMaterialWebServiceClient.java]: Nice usage of switch-case

File: myadmin-service-impls/integration-services/src/main/java/za/ac/unisa/myadmin/studymaterial/integration/services/StudyMaterialWebServiceClient.java [hunk 3]
```diff
@@ -99,22 +119,17 @@ private static int convertSemesterStringToId(String semester){
...
+		}
+		catch (ProcessingException pe){
+			throw new RuntimeException(SERVICE_UNAVAILBLE_MSG);
+		}
+		catch (Exception e) {
 			LOG.warn("Exception while trying to execute request", e);
 			if (response != null && response.getStatus() != 200) {
 				LOG.error("Failed : HTTP error code : " + response.getStatus());
-				throw new RuntimeException("Please note that UNISA study materials are not currently available for viewing. Access to the materials should be restored within an hour. UNISA apologizes for any inconvenience caused.");
+				throw new RuntimeException(SERVICE_UNAVAILBLE_MSG);
```
Comment (jannie-louwrens) [inline myadmin-service-impls/integration-services/src/main/java/za/ac/unisa/myadmin/studymaterial/integration/services/StudyMaterialWebServiceClient.java]: You probably meant to throw a new OperationFailedException instead

File: myadmin-webapps/myadmin-student-services-webapp/src/main/java/za/ac/unisa/myadmin/student/services/config/StudyMaterialServiceConfiguration.java [hunk 4]
```diff
@@ -32,17 +37,18 @@ public StudyMaterialService studyMaterialService(){
 		return studyMaterialService;
 	}
 
-	@Bean("studyMaterialFilterActiveDecorator")
-	public StudyMaterialService studyMaterialFilterActiveDecorator(){
+	@Bean(name = "studyMaterialServiceImplRestEndPoint")
+	public Server studyFeeQuotationServiceImplRestEndPoint() {
```
Comment (adrianhfoster) [inline myadmin-webapps/myadmin-student-services-webapp/src/main/java/za/ac/unisa/myadmin/student/services/config/StudyMaterialServiceConfiguration.java]: Copy paste error? studyFeeQuotationServiceImplRestEndPoint

File: myadmin-service-impls/student-services/src/main/java/za/ac/unisa/myadmin/student/services/impls/ModuleEnrolmentServiceImpl.java [hunk 5]
```diff
@@ -5,11 +5,7 @@
 import java.time.LocalDate;
 import java.time.ZoneId;
 import java.time.format.DateTimeFormatter;
-import java.util.ArrayList;
-import java.util.Collections;
-import java.util.Date;
-import java.util.List;
-import java.util.Optional;
+import java.util.*;
```
Comment (adrianhfoster) [inline myadmin-service-impls/student-services/src/main/java/za/ac/unisa/myadmin/student/services/impls/ModuleEnrolmentServiceImpl.java]: wildcard imports can cause ambiguity. Standard java packages are ok I guess.
I'm not bothered by it, Think I setup my IDE not to.

File: myadmin-webapps/myadmin-student-services-webapp/src/main/java/za/ac/unisa/myadmin/student/services/config/StudyMaterialServiceConfiguration.java (hunk 4 above)
Comment (trojanc) [reply myadmin-webapps/myadmin-student-services-webapp/src/main/java/za/ac/unisa/myadmin/student/services/config/StudyMaterialServiceConfiguration.java]: Fixed in follow up commit

**Pull request merged**